charts/microgateway/README.md.gotmpl

```diff
@@ -226,431 +236,399 @@ In case that the [Simple DSL configuration](#simple-dsl-configuration) does not
-## Environment variables
-With the Helm chart environment variables can be set.   
-These can then be used in the DSL configuration.   
-Below is an example of how to set environment variables and then use them in the DSL.   
-
-env-variables.yaml
-```
-config:
-  global:
-    env:
-      - name: WAF_CFG_OPERATIONALMODE
-        value: production
-      - name: WAF_CFG_LOGONLY
-        value: false
-```
-
-custom-values.yaml
-```
-config:
-  simple: 
-    mapping:
-      operationalMode: "@@WAF_CFG_OPERATIONALMODE@@"
-      denyRules:
-        logOnly: "@@WAF_CFG_LOGONLY@@"
-```
+## Security
+The following subchapters describes how use and deploy the Microgateway in a secure manner.
 
-## Probes
-In Kubernetes and Openshift probes are used to determine if a pod is ready to process requests or not.  
+### Store sensitive information in secrets
+Airlock Microgateway uses a few sensitive information which should be protected accordingly. The way to do that in Kubernetes or Openshift environments is, to store them in secrets.
+The following subchapters describe which information should be protected and how this can be achieved.
 
-### Liveness Probe
-The liveness Probe is used to find out when to restart a Pod.   
-If a liveness Probe fails, the Pod will restart.   
+#### Secure handling of license and passphrase
+It is possible to use the following parameters of this Helm chart to configure license and passphrase:
+* License: `config.generic.license`
+* Passphrase: `config.generic.passphrase`
 
-The Helm chart has already predefined a liveness probe.   
-If this does not work, it can be adjusted as follows.   
+The Helm chart itself creates a secret and configure the Microgateway to use it. While this is already secure, because it is stored as secrets, these information might end in a Git repo or somewhere too many people have access to.
+This is why it would be much better to create a secret containing license and passphrase with a different process.
 
-```
-livenessProbe:
-  initialDelaySeconds: 120
-```
+  The example below shows how to create a secret containing license and passphrase.
+  ```
+  kubectl create secret generic microgateway-secrets --from-file=license=<license_file> --from-file=passphrase=<passphrase_file>
+  ```
 
-The liveness Probe can be disabled with `livenessProbe.enabled=false`  
-Then the default of the environment on which the Microgateway is deployed is applied.   
+  Afterwards use this secret in the Helm chart configuration file.
+  custom-values.yaml
+  ```
+  config:
+    generic:
+      existingSecret: "microgateway-secrets"
+  ```
 
-### Readiness Probe
-The readiness Probe is used to set the status of a pod to Ready. This means that the pod is now ready to process requests.  
-If a readiness Probe fails, only the Ready status of the Pod is removed.   
+#### Credentials to pull image from docker registry
+The Microgateway image is published in a private Docker registry to which only granted accounts have access to.
```

Comment:
only granted accounts have access to. -->
only granted accounts have access.

Reply:
Done